src/target/riscv/riscv-013.c

```diff
@@ -5248,5 +5254,7 @@ static void riscv013_clear_abstract_error(struct target *target)
 		}
 	}
 	/* Clear the error status. */
-	dm_write(target, DM_ABSTRACTCS, DM_ABSTRACTCS_CMDERR);
+	if (dm_write(target, DM_ABSTRACTCS, DM_ABSTRACTCS_CMDERR) != ERROR_OK)
+		return ERROR_FAIL;
+	return ERROR_OK;
```

Comment:
Consider the following:
* The loop exits due to timeout.
* `dm_write` succeds.
* `ERROR_OK` is returned, although `abstractcs.cmderr` may be still set.

I would recommend replacing the loop with a call to `wait_for_idle`, as suggested [here](https://github.com/riscv/riscv-openocd/pull/964/files#diff-701864e2be3210275085c84b8f6fa7f9eb6171a2a18ca71638be42a155c08c63R5277), like so:
```
int res = wait_for_idle(...);
if (dm_write(target, DM_ABSTRACTCS, DM_ABSTRACTCS_CMDERR) != ERROR_OK)
  res = ERROR_FAIL;
return res;
```

Reply:
Done. Thanks.